src/sinks/util/tcp.rs

```diff
@@ -228,9 +233,9 @@ impl TcpSink {
         // If this returns `Poll::Pending` we know the connection is still
         // valid and the write will most likely succeed.
         let mut cx = Context::from_waker(noop_waker_ref());
-        match Pin::new(stream).poll_read(&mut cx, &mut [0u8; 1]) {
+        match Pin::new(stream).poll_read(&mut cx, &mut ReadBuf::new(&mut [0u8; 1])) {
             Poll::Ready(Err(error)) => ShutdownCheck::Error(error),
-            Poll::Ready(Ok(0)) => {
+            Poll::Ready(Ok(())) => {
```

Comment:
I think this should check for `buf.filled().is_empty()` here as described in [`poll_read`](https://docs.rs/tokio/1.3.0/tokio/io/trait.AsyncRead.html#tymethod.poll_read):

> If no data was read (`buf.filled().len()` is unchanged), it implies that `EOF` has been reached.

Reply:
Fixed in e21d0c6.